components/post_view/post_add_channel_member/post_add_channel_member.jsx

```diff
@@ -94,26 +106,57 @@ export default class PostAddChannelMember extends React.PureComponent {
                 return <span key={key}>{', '}</span>;
             }
 
+            if (this.state.expanded || usernames.length <= 3) {
+                return (
+                    <span>
+                        {
+                            usernames.map((username) => {
+                                return (
+                                    <AtMention
+                                        key={username}
+                                        mentionName={username}
+                                    />
+                                );
+                            }).reduce((acc, el, idx, arr) => {
+                                if (idx === 0) {
+                                    return [el];
+                                } else if (idx === arr.length - 1) {
+                                    return [...acc, andSeparator(idx), el];
+                                }
+
+                                return [...acc, commaSeparator(idx), el];
+                            }, [])
+                        }
+                    </span>
+                );
+            }
+            const otherUsers = [...usernames];
+            const firstUserName = otherUsers.shift();
+            const lastUserName = otherUsers.pop();
             return (
                 <span>
-                    {
-                        usernames.map((username) => {
-                            return (
-                                <AtMention
-                                    key={username}
-                                    mentionName={username}
-                                />
-                            );
-                        }).reduce((acc, el, idx, arr) => {
-                            if (idx === 0) {
-                                return [el];
-                            } else if (idx === arr.length - 1) {
-                                return [...acc, andSeparator(idx), el];
-                            }
-
-                            return [...acc, commaSeparator(idx), el];
-                        }, [])
-                    }
+                    <AtMention
+                        key={firstUserName}
+                        mentionName={firstUserName}
+                    />
+                    {commaSeparator(1)}
+                    <a
+                        id='post_body_other_users_link'
```

Comment:
Is it possible for this component to be rendered in the DOM more than once?
If so, having 2 elements with this `id` is considered bad form.
If not it's alright but I would avoid using `id` here if you don't need it.

Reply:
the main reason I added it was to discern between the other link within this component for testing... it is possible for this component to be on the page multiple times, would a class be preferable for this use case @devinbinnie?

Reply:
Yes definitely, or you can add a custom data tag like we use for Cypress.
Normally it's called `data-testid`.